src/kademlia/node.cpp

```diff
@@ -220,7 +220,10 @@ void node::bootstrap(std::vector<udp::endpoint> const& nodes
 
 	for (auto const& n : nodes)
 	{
-		if (n.protocol() != protocol()) continue;
+#if !TORRENT_USE_IPV6
+		if (n.protocol() == udp::v6()) continue;
```

Comment:
if you don't concatenate ``nodes6``, presumably this wouldn't happen, right?
I mean, maybe you could have an ifdef to not add in the v6 nodes in the first place, if ``TORRENT_USE_IPV6`` wasn't set.